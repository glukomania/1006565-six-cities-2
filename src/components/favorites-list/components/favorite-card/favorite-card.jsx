import {Operations} from "../../../../store/reducer";
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';

const FavoriteCard = (props) => {

  const bookmarkRef = React.createRef();

  const bookmarkClickHandle = () => {
    bookmarkRef.current.classList.remove(`place-card__bookmark-button--active`);
    props.setFavorite(props.favorite.id, 0);
    bookmarkRef.current.disabled = true;
  };

  const setAddress = () => {
    return `/offer/${props.favorite.id}`;
  };

  return <article className="favorites__card place-card">
    <div className="favorites__image-wrapper place-card__image-wrapper">
      <Link to={setAddress()}>
        <img className="place-card__image" src={props.favorite.preview_image} width="150" height="110" alt="Place image" />
      </Link>
    </div>
    <div className="favorites__card-info place-card__info">
      <div className="place-card__price-wrapper">
        <div className="place-card__price">
          <b className="place-card__price-value">&euro;{props.favorite.price}</b>
          <span className="place-card__price-text">&#47;&nbsp;night</span>
        </div>
        <button className="place-card__bookmark-button place-card__bookmark-button--active button" ref={bookmarkRef} type="button" onClick={bookmarkClickHandle}>
          <svg className="place-card__bookmark-icon" width="18" height="19">
            <use xlinkHref="#icon-bookmark"></use>
          </svg>
          <span className="visually-hidden">In bookmarks</span>
        </button>
      </div>
      <div className="place-card__rating rating">
        <div className="place-card__stars rating__stars">
          <span style={{width: props.favorite.rating * 20 + `%`}}></span>
          <span className="visually-hidden">Rating</span>
        </div>
      </div>
      <h2 className="place-card__name">
        <Link to={setAddress()}>{props.favorite.title}</Link>
      </h2>
      <p className="place-card__type">{props.favorite.type}</p>
    </div>
  </article>;
};

FavoriteCard.propTypes = {
  favorite: PropTypes.object,
  setFavorite: PropTypes.func,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  setFavorite: state.setFavorite,
});

const mapDispatchToProps = {
  setFavorite: (id, status) => Operations.setFavorite(id, status),
};


export default connect(mapStateToProps, mapDispatchToProps)(FavoriteCard);
