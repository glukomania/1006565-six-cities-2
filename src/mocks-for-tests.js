// export const offers = [
//   {
//     id: 1,
//     city: {
//       name: `Amsterdam`,
//       location: {
//         latitude: `52.369553943508`,
//         longitude: `4.85309666406198`,
//       }
//     },
//     name: `Beautiful & luxurious apartment at great location`,
//     images: [`img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/apartment-01.jpg`],
//     price: 80,
//     rating: 80,
//     type: `Private room`,
//     propFeatures: {entire: `Entire place`, bedroom: 3, adults: 3},
//     insideItems: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
//     host: {name: `William`, status: `Free`, avatar: `img/avatar-angelina.jpg`},
//     description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
//     location: {
//       latitude: `52.369553943508`,
//       longitude: `4.85309666406198`
//     },
//   },
//   {
//     id: 2,
//     city: {
//       name: `Amsterdam`,
//       location: {
//         latitude: `52.369553943508`,
//         longitude: `4.85309666406198`,
//       }
//     },
//     name: `Beautiful & luxurious apartment at great location`,
//     images: [`img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/apartment-01.jpg`],
//     price: 80,
//     rating: 80,
//     type: `Private room`,
//     propFeatures: {entire: `Entire place`, bedroom: 3, adults: 3},
//     insideItems: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
//     host: {name: `William`, status: `Free`, avatar: `img/avatar-angelina.jpg`},
//     description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
//     location: {
//       latitude: `52.369553943508`,
//       longitude: `4.85309666406198`
//     },
//   },
//   {
//     id: 3,
//     city: {
//       name: `Berlin`,
//       location: {
//         latitude: `55.1`,
//         longitude: `4.1`,
//       }
//     },
//     name: `Beautiful & luxurious apartment at great location`,
//     images: [`img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/apartment-01.jpg`],
//     price: 80,
//     rating: 80,
//     type: `Private room`,
//     propFeatures: {entire: `Entire place`, bedroom: 3, adults: 3},
//     insideItems: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
//     host: {name: `William`, status: `Free`, avatar: `img/avatar-angelina.jpg`},
//     description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
//     location: {
//       latitude: `55.1`,
//       longitude: `4.1`
//     },
//   },
// ];


export const offers = [
  {
    'id': 1,
    'city': {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    'preview_image': `img/1.png`,
    'images': [`img/1.png`, `img/2.png`],
    'title': `Beautiful & luxurious studio at great location`,
    'is_favorite': false,
    'is_premium': false,
    'rating': 4.8,
    'type': `apartment`,
    'bedrooms': 3,
    'max_adults': 4,
    'price': 120,
    'goods': [`Heating`, `Kitchen`, `Cable TV`],
    'host': {
      'id': 3,
      'is_pro': true,
      'name': `Angelina`,
      'avatar_url': `img/1.png`
    },
    'description': `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    'location': {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  }
];


export const feedbacks = [
  {
    id: 1,
    user: {
      'id': 4,
      'is_pro': false,
      'name': `Max`,
      'avatar_url': `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
  }
];
