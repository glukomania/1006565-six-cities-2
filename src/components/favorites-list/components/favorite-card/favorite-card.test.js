import renderer from 'react-test-renderer';
import {FavoriteCard} from './favorite-card';
import {offers} from '../../../../mocks-for-tests';

jest.mock(`react-router-dom`, () => ({
  Link: () => null
}));

it(`FavoriteCard is displayed correctly`, () => {

  const props = {
    isInner: true,
    isAuthorized: false,
    favorite: offers[0],
  };

  const favoriteCard = renderer
    .create(<FavoriteCard {...props} />)
  .toJSON();

  expect(favoriteCard).toMatchSnapshot();
});
