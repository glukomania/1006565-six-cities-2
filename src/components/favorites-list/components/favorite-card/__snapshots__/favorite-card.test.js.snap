// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteCard is displayed correctly 1`] = `
<article
  className="favorites__card place-card"
>
  <div
    className="favorites__image-wrapper place-card__image-wrapper"
  />
  <div
    className="favorites__card-info place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          120
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button place-card__bookmark-button--active button"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          In bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "96%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    />
    <p
      className="place-card__type"
    >
      apartment
    </p>
  </div>
</article>
`;
